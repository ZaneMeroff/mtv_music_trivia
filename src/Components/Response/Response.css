.response-contianer {
  align-items: center;
  display: flex;
  flex-direction: column;
  height: 70%;
  justify-content: center;
  text-align: center;
}

.response-text {
  color: #fff200;
  font-size: 300%;
  transform: rotate(-20deg);
}

@media only screen and (max-width: 490px) {

  .response-text {
    font-size: 200%;
  }

}
