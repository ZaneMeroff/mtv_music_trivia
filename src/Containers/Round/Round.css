.answer-button,
.question-container {
  display: flex;
  align-items: center;
  height: 40px;
  justify-content: center;
  margin: 2% 0% 2% 0%;
  width: 65%;
}

.answer-button {
  border: none;
  font-size: 110%;
}

.answer-button {
  background-color: #ff64de;
  border-radius: 20px;
}

.answer-button:hover {
  cursor: pointer;
  border: 5px solid #fff200;
  color: #FFF;
}

.question-container {
  margin-top: 30px;
  height: auto;
  width: 90%;
}

.question-text {
  font-size: 120%;
  padding: 1%;
  text-align: center;
}

.submit-answer-button {
  background-color: #fff200;
  border-radius: 20px;
  border: none;
  font-size: 110%;
  height: 40px;
  text-transform: uppercase;
  width: 65%;
}

.submit-answer-button:hover {
  background-color: #ff64de;
  color: #FFF;
  cursor: pointer;
}

.round {
  align-items: center;
  display: flex;
  flex-direction: column;
  justify-content: center;
}

.active-button {
  border: 5px solid #fff200;
  color: #FFF;
}

.round-error-message {
  color: #fff200;
  font-size: 100%;
}

@media only screen and (max-width: 490px) {

  .answer-button,
  .question-container {
    height: 40px;
    font-size: 90%;
  }

  .submit-answer-button {
    height: 40px;
    font-size: 90%;
  }

  .question-text {
    font-size: 100%;
  }

}
