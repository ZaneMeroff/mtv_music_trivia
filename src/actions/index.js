export const getTriviaData = triviaData => ({
  type: 'GET_TRIVIA_DATA',
  triviaData
});

export const correctQuestions = question => ({
  type: 'SAVE_CORRECT_QUESTION',
  question
});

export const incorrectQuestions = question => ({
  type: 'SAVE_INCORRECT_QUESTION',
  question
});
