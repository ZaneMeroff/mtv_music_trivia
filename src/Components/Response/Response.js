import React from 'react';
import './Response.css';
import PropTypes from 'prop-types';

const Response = props => {
    return (
      <div className='response-contianer'>
        <p className='response-text'>{props.text}</p>
      </div>
    )
}

Response.propTypes = {
  text: PropTypes.string,
}

export default Response;
