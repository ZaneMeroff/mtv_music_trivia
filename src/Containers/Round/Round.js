import React, { Component } from 'react';
import './Round.css';

class Round extends Component {
  constructor(props) {
    super(props);
    this.state = { selectedAnswer: null }
  }

  upDateSelectedAnswer = answer => {
    this.setState({selectedAnswer: answer })
  }

  submitAnswer = () => {


  }

  render() {
    return (
      <div className='round'>
        <div className='question-container'>
          <p className='question-text'>From which country did the song Gangnam Style originate from?</p>
        </div>
        <button onClick={e => this.upDateSelectedAnswer(e.target.value)} value='South Korea' className={this.state.selectedAnswer ? 'answer-button active-button' : 'answer-button'}>South Korea</button>
        <button onClick={e => this.upDateSelectedAnswer(e.target.value)} value='Japan' className='answer-button'>Japan</button>
        <button onClick={e => this.upDateSelectedAnswer(e.target.value)} value='South Korea' className='answer-button'>North Korea</button>
        <button onClick={e => this.upDateSelectedAnswer(e.target.value)} value='China' className='answer-button'>China</button>
        <button className='submit-answer-button' onClick={this.submitAnswer}>SUBMIT ANSWER</button>
      </div>
    )

  }

}

export default Round;

// From which country did the song Gangnam Style originate from?
