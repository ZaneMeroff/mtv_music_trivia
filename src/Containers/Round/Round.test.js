import React from 'react';
import ReactDOM from 'react-dom';
import Response from '../../Components/Response/Response';
import { Round, mapDispatchToProps, mapStateToProps } from './Round';
import { shallow } from 'enzyme';
import { correctQuestions, incorrectQuestions} from '../../actions/index';

describe('Round', () => {

  let wrapper, mockProps, mockState;

  beforeEach(() => {
    mockState = {
      selectedAnswer: null,
      errorMessage: '',
      rightORwrong: null,
      counter: 0
    }
    mockProps = {
      addToCorrectQuestions: jest.fn(),
      addToIncorrectQuestions: jest.fn(),
    }
    wrapper = shallow(<Round
      triviaData={{
        question: 'What is music?',
        correct_answer: 'life'}}
      {...mockProps}
      />);
  })

  describe('Round container/component', () => {

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should start off with a default state', () => {
      const expectedState = {
        selectedAnswer: null,
        errorMessage: '',
        rightORwrong: null,
        counter: 0
      }
      expect(wrapper.state()).toEqual(expectedState)
    })

    it('should update state when updateSelectedAnswer is called', () => {
      const mockAnswer = 'The Beatles';
      const expectedAnswer = 'The Beatles';
      wrapper.instance().updateSelectedAnswer(mockAnswer)
      expect(wrapper.state('selectedAnswer')).toEqual(expectedAnswer)
    });

    describe('submitAnswer', () => {

      it('should update state is no selected answer', () => {
        const expectedState = {
          selectedAnswer: null,
          errorMessage: 'You Must Select an Answer!',
          rightORwrong: null,
          counter: 0
        }
        wrapper.instance().submitAnswer()
        expect(wrapper.state()).toEqual(expectedState)
      });

      it('should call addToCorrectQuestions, reset state, and resetForNextRound if user selects right answer', () => {
        wrapper = shallow(<Round
          triviaData={[{
            question: 'What is music?',
            correct_answer: 'life',
            all_answers: ['life', 'fun', 'great', 'awesome']
          }]}
          {...mockProps}
          />);
        const expectedData = {
          question: 'What is music?',
          correct_answer: 'life',
          all_answers: ['life', 'fun', 'great', 'awesome']
        }
        wrapper.instance().resetForNextRound = jest.fn()
        wrapper.setState({selectedAnswer: 'life'})
        wrapper.instance().submitAnswer()
        expect(mockProps.addToCorrectQuestions).toHaveBeenCalledWith(expectedData)
        expect(wrapper.state('rightORwrong')).toEqual(true)
        expect(wrapper.instance().resetForNextRound).toHaveBeenCalled()
      });

      it('should call addToIncorrectQuestions, setState, and resetForNextRound if user selects wrong answer', () => {
        wrapper = shallow(<Round
          triviaData={[{
            question: 'What is music?',
            correct_answer: 'life',
            all_answers: ['life', 'fun', 'great', 'awesome']
          }]}
          {...mockProps}
          />);
        const expectedData = {
          question: 'What is music?',
          correct_answer: 'life',
          all_answers: ['life', 'fun', 'great', 'awesome'],
          your_answer: 'great'
        }
        wrapper.instance().resetForNextRound = jest.fn()
        wrapper.setState({selectedAnswer: 'great'})
        wrapper.instance().submitAnswer()
        expect(mockProps.addToIncorrectQuestions).toHaveBeenCalledWith(expectedData)
        expect(wrapper.state('rightORwrong')).toEqual(false)
        expect(wrapper.instance().resetForNextRound).toHaveBeenCalled()
      });

    });

    it('should update state when resetForNextRound is called', () => {
      wrapper.instance().resetForNextRound()
      expect(wrapper.state('errorMessage')).toEqual('')
    });

    it('should call updateSelectedAnswer when an answer button is clicked', () => {
      wrapper = shallow(<Round
        triviaData={[{
          question: 'What is music?',
          correct_answer: 'life',
          all_answers: ['life', 'fun', 'great', 'awesome']
        }]}
        {...mockProps}
        />);
      const mockEvent = {
      target: {
        name: 'mockName',
        value: 'mockValue'
      }}
      wrapper.instance().updateSelectedAnswer = jest.fn();
      wrapper.instance().forceUpdate()
      wrapper.find('.answer-button').at(0).simulate('click', mockEvent)
      expect(wrapper.instance().updateSelectedAnswer).toHaveBeenCalled()
    });

    it('should call submitAnswer when submit answer button is clicked', () => {
      wrapper.instance().submitAnswer = jest.fn();
      wrapper.instance().forceUpdate()
      wrapper.find('.submit-answer-button').simulate('click')
      expect(wrapper.instance().submitAnswer).toHaveBeenCalled()
    });

  })

  describe('displayRightOrWrong', () => {

    it('should return Response component with RIGHT! if rightORwrong is true', () => {
      const mockRightOrWrong = true;
      const result = wrapper.instance().displayRightOrWrong(mockRightOrWrong)
      const expected = <Response text='RIGHT!'/>
      expect(result).toEqual(expected)
    });

    it('should return Response component with WRONG! if rightORwrong is false', () => {
      const mockRightOrWrong = false;
      const result = wrapper.instance().displayRightOrWrong(mockRightOrWrong)
      const expected = <Response text='WRONG!'/>
      expect(result).toEqual(expected)
    });

  })

})

describe('mapStateToProps', () => {

  it('should return an array of trivia data', () => {
    const triviaData = { triviaData: [{
      question: 'Which band is best?',
      correct_answer: 'The Beatles',
      incorrect_answers: ['The Who', 'Rush']
    }]}
    const expected = { triviaData: [{
      question: 'Which band is best?',
      correct_answer: 'The Beatles',
      incorrect_answers: ['The Who', 'Rush']
    }]}
    const mappedProps = mapStateToProps(triviaData)
    expect(mappedProps).toEqual(expected)
  })

})

describe('mapDispatchToProps', () => {

  it('should dispatch correctQuestions with round data', () => {
    const mockDispatch = jest.fn();
    const triviaData = {
      question: 'Which band is best?',
      correct_answer: 'The Beatles',
      incorrect_answers: ['The Who', 'Rush']
    }
    const actionToDispatch = correctQuestions(triviaData)
    const mappedProps = mapDispatchToProps(mockDispatch)
    mappedProps.addToCorrectQuestions(triviaData)
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
  })

  it('should dispatch incorrectQuestions with round data', () => {
    const mockDispatch = jest.fn();
    const triviaData = {
      question: 'Which band is best?',
      correct_answer: 'The Beatles',
      incorrect_answers: ['The Who', 'Rush']
    }
    const actionToDispatch = incorrectQuestions(triviaData)
    const mappedProps = mapDispatchToProps(mockDispatch)
    mappedProps.addToIncorrectQuestions(triviaData)
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
  })

})
