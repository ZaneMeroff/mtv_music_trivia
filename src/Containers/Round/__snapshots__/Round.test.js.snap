// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Round Round container/component should match the snapshot 1`] = `
<Response
  text="loading..."
/>
`;
